docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,118 @@
+  
+NOTE:  As the services now self seed and deployment specific changes can be made via environment overrides, it will no longer be necessary to have a Docker profile configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.  It will still be possible for users to use the profile mechanism to specify a Docker configuration, but it will no longer be required and not the recommended approach to providing Docker container specific configuration.
+
+### Overrides
+Environmental variables override configuration or command line option values in the following ways
+- Environmental variables override local configuration; that is when configuration for a service is obtained from the local config file
+- Environmental variables override configuration values as they are pushed (self seeded) into the configuration service (Consul).  This override only occurs once (as the values are pushed / seeded into Consul from the service).  Once pushed, configuration values are used from Consul and any environmental value is ignored unless the -o/--overwrite flag is on.  The name of the environmental variable must match the path names in Consul.
+- Environmental variables can override command line options (except the -o overwrite and -r registry command line options). 
+
+NOTES:
+- Environmental variables do not override any local configuration; that is when configuration for a service is obtained from the local config file but not from the configuration service, the environmental variables are ignored.
+- Environmental variable overrides remove the need to change the "docker" profile in the res/docker/configuration.toml files - Allowing removal of 50% of the existing configuration.toml files.
+
+All values overriden get logged (indicating which configuration value or op param and the new value).  
+
+## Decision 
+
+These features have been implemented (with some minor changes to be done) for consideration here:  https://github.com/edgexfoundry/go-mod-bootstrap/compare/master...lenny-intel:SelfSeed2
+
+The implementation for self-seeding services and environmental overrides is already implemented (for Fuji) per this document in the application services and device services (and instituted in the SDKs of each).
+
+## Backward compatibility
+Several aspects of this ADR contain backward compatibility issues for the device service and application service SDKs.  Therefore, for the upcoming minor release, the following guidelines and expections are added to provide for backward compatibility.
+
+- --registry=<url> for Device SDK
+        
+If used with URL , assume previous version and use URL for --configProvider option and default to http as DS SDK does today. Also use URL for registry connection.
+
+- --registry (no ‘=’) and w/o --configProvider for both SDKs
```

Comment:
Isn't this for all services (e.g. app, core-*, device and support-*)?

Reply:
Not for core since the actual service images/snaps will be created for Geneva with the new options.